### ISSUE TITLE
Submit 3.1-based app to Apple App Store

### ISSUE THREAD

**Issue by freedomtan** [mlcommons/mobile_app_open#780]
Let's check what need to be done before submitting to App Store

- [ ] requirement 1
- [ ] requirement 2

And then submit it.

**Comment by freedomtan** [comment]
ask @nathanw-mlc for the App Store account

**Comment by nathanw-mlc** [comment]
Anh already has access to the App Store accounts. Will anyone else need access?

**Comment by RSMNYS** [comment]
Hi guys! We need screenshots with such sizes:

<img width="306" alt="Screenshot 2023-09-13 at 14 39 06" src="https://github.com/mlcommons/mobile_app_open/assets/17128789/71e98109-ea10-46c7-a9bc-89d3d6504f3f">

Based on this: https://developer.apple.com/help/app-store-connect/reference/screenshot-specifications

6.5":

iPhone 14 Plus
iPhone 13 Pro Max
iPhone 12 Pro Max
iPhone 11 Pro Max
iPhone 11
iPhone XS Max
iPhone XR

5.5":

iPhone 8 Plus
iPhone 7 Plus
iPhone 6s Plus

Unfortunately we can't make the screenshots from the simulators as it has Debug label on the screen, and such screenshots are not acceptable. So we need to find these devices to prepare the screenshots for the submission.

**Comment by RSMNYS** [comment]
Actually it's possible to remove that label for the debug flutter mode. 
![Simulator Screenshot - iPhone 14 Pro Max - 2023-09-13 at 15 00 42](https://github.com/mlcommons/mobile_app_open/assets/17128789/1075d318-d26c-4935-8611-c5b974e04a15)

Let me know if I should prepare all screenshots and upload to the App Store. My email is: Sergij.r@scopicsoftware.com.

**Comment by nathanw-mlc** [comment]
@RSMNYS What Apple ID email address should I use to grant you access to the App Store? If you don't want to post it publicly in this thread, you can email systems@mlcommons.org.

**Comment by RSMNYS** [comment]
Sergij.r@scopicsoftware.com. Thank you

**Comment by nathanw-mlc** [comment]
Thanks, invitation sent.

**Comment by RSMNYS** [comment]
> @RSMNYS What Apple ID email address should I use to grant you access to the App Store? If you don't want to post it publicly in this thread, you can email [systems@mlcommons.org](mailto:systems@mlcommons.org).

@nathanw-mlc please change my role to the app manager, with the developer role I'm not able to upload the screenshots. 

Thanks

**Comment by nathanw-mlc** [comment]
@RSMNYS, I've updated your role as requested.

**Comment by RSMNYS** [comment]
<img width="1285" alt="Screenshot 2023-09-18 at 21 42 30" src="https://github.com/mlcommons/mobile_app_open/assets/17128789/a4b8101c-16c4-4511-8f87-d1213ac2ba68">
<img width="1285" alt="Screenshot 2023-09-18 at 21 42 49" src="https://github.com/mlcommons/mobile_app_open/assets/17128789/a826904d-bfa0-499a-bdff-1b9692833c0a">

**Comment by nathanw-mlc** [comment]
Do we want screenshots with the blue background of the official release, since presumably that's what people will be downloading? If you need, I can provide them.

**Comment by anhappdev** [comment]
https://developer.apple.com/app-store/app-privacy-details/

**Comment by freedomtan** [comment]
- privacy policy statement: how user-generated data are treated (where are they stored, how they are to be used by MLCommons), etc.
- app description: short and long description 

@TheKanter and @peladodigital

**Comment by freedomtan** [comment]
@Mostelk let's try to discuss the privacy policy statement issue in the working group policy meeting.

**Comment by freedomtan** [comment]
|iphone 15 Pro Max| iPhone 14 Pro | iPhone 13|
|---------------------------|----------------------|----------------|
|<img src="https://github.com/mlcommons/mobile_app_open/assets/3395998/f2bd246b-aa29-42d2-99ec-8cca46d09f16" width="200"  />|<img src="https://github.com/mlcommons/mobile_app_open/assets/3395998/c18e59ac-6462-41a7-b2fe-4348d29f2be6" width="200"  />|<img src="https://github.com/mlcommons/mobile_app_open/assets/3395998/7d799577-41fc-4738-825a-98db492dd541" width="200"  />|

**Comment by freedomtan** [comment]
An example [`result.json`](https://drive.google.com/file/d/1ObI5sK1mOVQkZLassf3GCsNBFvjTkmFP/view?usp=share_link) file on iPhone 13.

Currently, besides benchmark numbers, what we collected in the app are:

```json
{
  ...,
  "environment_info": {
    "platform": "ios",
    "value": {
      "android": null,
      "ios": {
        "os_version": "Version 17.0.1 (Build 21A340)",
        "model_code": "iPhone14,5",
        "model_name": "iPhone 13",
        "soc_name": "Apple A15"
      },
      "windows": null
    }
  },
  "build_info": {
    "version": "3.1.0",
    "build_number": "1",
    "build_date": "2023-09-22T16:56:43.134131",
    "official_release_flag": false,
    "dev_test_flag": false,
    "backend_list": [
      "libcoremlbackend",
      "libtflitebackend"
    ],
    "git_branch": "ndk_25_quick_hack",
    "git_commit": "73942b6a1a725972d6f9f6390dd4315f9c663a0b",
    "git_dirty_flag": true
  }
}
```

**Comment by peladodigital** [comment]
@RSMNYS Please go ahead with these for the configuration of the store. We can go with this and then later add if needed

**Comment by RSMNYS** [comment]
> @RSMNYS Please go ahead with these for the configuration of the store. We can go with this and then later add if needed

@peladodigital I think you've missed the original request. Let me share it again. To submit the app to the store we need to check and fill the next points:

- privacy policy statement: how user-generated data are treated (where are they stored, how they are to be used by MLCommons), etc. https://developer.apple.com/app-store/app-privacy-details/
- app description: short and long description
- test accounts: pre-created accounts for Apple to test? or ask them to create accounts in the App?
we have to create accounts for them before submission

@peladodigital  can you help with this please? 

Here is the link to the app store: https://appstoreconnect.apple.com/apps/1629525778/appstore/privacy

**Comment by freedomtan** [comment]
@TheKanter and @peladodigital  Please help provide the privacy policy statement, c.f. https://developer.apple.com/app-store/app-privacy-details/.

@RSMNYS please pick one from TestFlight.

**Comment by TheKanter** [comment]
I'm working to get one generated for us. Do you have any examples I can
share with our attorneys? Ideally a simple one.

**Comment by anhappdev** [comment]
> I'm working to get one generated for us. Do you have any examples I can share with our attorneys? Ideally a simple one.

@TheKanter You may use one of the templates from:
https://termly.io/products/privacy-policy-generator/

**Comment by freedomtan** [comment]
maybe we need both "terms of use" / "EULA" and "privacy policy"

**Comment by TheKanter** [comment]
In on it with our lawyers.

**Comment by freedomtan** [comment]
@RSMNYS and @anhappdev  can we submit the iOS app to mac app store? Or we need native macOS UI?

**Comment by anhappdev** [comment]
> @RSMNYS and @anhappdev can we submit the iOS app to mac app store? Or we need native macOS UI?

Yes, we can but the app will work only on Mac computers with Apple silicon (M1, M2, ...)
https://support.apple.com/en-vn/guide/app-store/fird2c7092da/mac

**Comment by TheKanter** [comment]
That’s fine to only test Apple silicon. We also should have TOU and privacy
policy soon.

**Comment by freedomtan** [comment]
Let's enable macOS compatibility for TestFlight, then we can test the app on both iOS and macOS (apple silicon machines)

**Comment by TheKanter** [comment]
I have a MacBook Air we can use!

**Comment by anhappdev** [comment]
The app should be already available on TestFlight for Mac (https://apps.apple.com/us/app/testflight/id899247664)

**Comment by freedomtan** [comment]
@@RSMNYS please check if you got the term of use and privacy policy docs. If yes, let's try to make this into the app store.

@freedomtan to check if those docs are available to @RSMNYS and @anhappdev.

**Comment by freedomtan** [comment]
open source:
  TensorFlow: include TensorFlow's apache license
  models: apache license should be enough

@freedomtan to crease an issue to track this and check open source third-party libraries we used. Check other benchmark programs.

---

vendor spec (for Android):
 * SNPE:  binary libraries included, should be OK to re-distributed,  @mohitmundhragithub will help check that
 * Samsung: binaries should be fine, @Mostelk

**Comment by anhappdev** [comment]
@RSMNYS Since we have a new design for v4.0, can you please update the screenshots in the App Store Connect?

**Comment by RSMNYS** [comment]
@anhappdev are those screenshots correct? I'm using the v4.0 branch, official_build, but see a bunch of UI issues as below. Where we use black text on blue background, and other components also looks wrong and visibility is bad. Please let me know.

![Simulator Screenshot - iPhone 15 Pro Max - 2024-05-31 at 18 10 52](https://github.com/mlcommons/mobile_app_open/assets/17128789/24df4eb2-e17a-44a5-bfd7-291f24fda90f)

![Simulator Screenshot - iPhone 15 Pro Max - 2024-05-31 at 18 05 40](https://github.com/mlcommons/mobile_app_open/assets/17128789/61a9507e-3570-4a2e-b3bf-ca7d7c015ad6)
![Simulator Screenshot - iPhone 15 Pro Max - 2024-05-31 at 18 05 45](https://github.com/mlcommons/mobile_app_open/assets/17128789/682c6138-6085-4e3b-b4bc-9315b45301c5)
![Simulator Screenshot - iPhone 15 Pro Max - 2024-05-31 at 18 15 46](https://github.com/mlcommons/mobile_app_open/assets/17128789/2494bb46-b62c-41d3-890a-f486f9f74356)

![Simulator Screenshot - iPhone 15 Pro Max - 2024-05-31 at 18 05 24](https://github.com/mlcommons/mobile_app_open/assets/17128789/b7bd4ca5-b7cd-4fb6-8173-e4be0aae2f40)

**Comment by anhappdev** [comment]
> @anhappdev are those screenshots correct? I'm using the v4.0 branch, official_build, but see a bunch of UI issues as below. Where we use black text on blue background, and other components also looks wrong and visibility is bad. Please let me know.

I am not sure why your build looks different, maybe you used a different Flutter SDK version?
For iOS, can you use the version 4.0 (build 125) from TestFlight? The UI looks correct on my iPhone with the build from TestFlight.

**Comment by RSMNYS** [comment]
You're right @anhappdev. Now have everything looks good. By the way, we can use the fvm to manage the flutters versions, and lock the flutter version for the project. So everyone can use the same version of the flutter sdk. What do you think?

**Comment by anhappdev** [comment]
> You're right @anhappdev. Now have everything looks good. By the way, we can use the fvm to manage the flutters versions, and lock the flutter version for the project. So everyone can use the same version of the flutter sdk. What do you think?

We have already defined the Flutter SDK version in the `pubspec.yml` file. And we don't change the Flutter SDK version often, so I am not sure if we need yet another tool.

**Comment by anhappdev** [comment]
@nathanw-mlc Can you help resolve this one?
> The items below are required to start the review process:
Under the Digital Services Act, you must provide and verify information regarding your account. [Learn More](https://developer.apple.com/help/app-store-connect/manage-compliance-information/manage-eu-digital-services-act-compliance-information)

**Comment by RSMNYS** [comment]
> @RSMNYS Please also upload screenshots for iPad.
> 
> > The items below are required to start the review process:
> > You must upload a screenshot for 12.9-inch iPad Pro (2nd generation) displays.
> > You must upload a screenshot for 13-inch iPad displays.

done

**Comment by freedomtan** [comment]
close it for now.
